### PR TITLE
Optical sat constellation design

### PR DESCRIPTION
**Pull request by ShamilBiktimirov**
Vectorized functions for precession rates and nodal period.

Function for calculating fundamental interval

### CODE REVIEW

**Comment by ShamilBiktimirov** [comment]
I think it is relevant to add the function on Sf calculation for different latitudes

File: orbitDesign/calcFundamentalGroundTrackInterval.m [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function sF = calcFundamentalGroundTrackInterval(oe)
+
+    % orbit is assumed as circular
+
+    [raanDot, ~, ~] = calcSecularPrecessionRates(oe(1, :), zeros(1, size(oe, 2)), oe(3, :));
+    periodNodal     = calcDraconicPeriod([oe(1, :); zeros(1, size(oe, 2)); oe(3, :)]);
```
Comment (FatimaAlnaqbi1998) [inline orbitDesign/calcFundamentalGroundTrackInterval.m:6]: I think you deleted this function and replace it with calcPeriodNodal, so this should be updated

File: orbitDesign/calcSsoRgtOrbitSma.m [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    % Author: Shamil Biktimirov
+
+    sma = [];
+    period_greenwich_nodal = 2 * pi / (Consts.omegaEarth - Consts.earthMeanMotion);
+    k_revpday = k_rev2rep / k_day2rep;
+    n_assumption = k_revpday * Consts.omegaEarth;
+    a_assumption = (Consts.muEarth * (1 / n_assumption) ^ 2) ^ (1 / 3);
+
+    a_step = 0.01;  % m
+    delta_a = [-50e3:a_step:50e3];
```
Comment (FatimaAlnaqbi1998) [inline orbitDesign/calcSsoRgtOrbitSma.m:15]:
```suggestion
    delta_a = [-50e3 : a_step : 50e3];
```

File: orbitDesign/findOrbitRgtProperties.m [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
     %% Calculates Earth and satellite nodal periods and its ratio
 
-    pSatNodal = calcDraconicPeriod([a, 0, i]);
+    pSatNodal = calcDraconicPeriod([a; 0; i]);
```
Comment (FatimaAlnaqbi1998) [inline orbitDesign/findOrbitRgtProperties.m:19]: I think you deleted this function and replace it with calcPeriodNodal, so this should be updated

**Review by FatimaAlnaqbi1998 — COMMENTED** [review]
three minor comments

File: orbitDesign/findOrbitRgtProperties.m (hunk 3 above)
Comment (ShamilBiktimirov) [reply orbitDesign/findOrbitRgtProperties.m]: Improved

File: orbitDesign/calcFundamentalGroundTrackInterval.m (hunk 1 above)
Comment (ShamilBiktimirov) [reply orbitDesign/calcFundamentalGroundTrackInterval.m]: Thanks, noted. I went through the whole repository and updated it accordingly

**Review by FatimaAlnaqbi1998 — APPROVED** [review]
approved

**Pull request merged**